op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -59,25 +48,58 @@ var (
 		Usage:    "Path to write the rollup config JSON file, to be provided to op-node with the 'rollup.config' flag",
 		Required: true,
 	}
-	dbPathFlag = &cli.StringFlag{
-		Name:     "db-path",
-		Usage:    "Path to the Celo database, not including the `celo/chaindata` part",
+	oldDBPathFlag = &cli.PathFlag{
+		Name:     "old-db",
+		Usage:    "Path to the old Celo chaindata",
+		Required: true,
+	}
+	newDBPathFlag = &cli.PathFlag{
+		Name:     "new-db",
+		Usage:    "Path to write migrated Celo chaindata",
 		Required: true,
 	}
-	dryRunFlag = &cli.BoolFlag{
-		Name:  "dry-run",
-		Usage: "Dry run the upgrade by not committing the database",
+	batchSizeFlag = &cli.Uint64Flag{
+		Name:  "batch-size",
+		Usage: "batch size to use for block migration",
+		Value: 50000,
+	}
+	memoryLimitFlag = &cli.Int64Flag{
+		Name:  "memory-limit",
+		Usage: "Memory limit in MB",
+		Value: 7500,
+	}
+	clearAllFlag = &cli.BoolFlag{
+		Name:  "clear-all",
+		Usage: "Use this to start with a fresh new db, deleting all data including ancients. CAUTION: Re-migrating ancients takes time.",
+	}
+	keepNonAncientsFlag = &cli.BoolFlag{
+		Name:  "keep-non-ancients",
+		Usage: "CAUTION: Not recommended for production. Use to keep all data in the new db as is, including any partially migrated non-ancient blocks and state data. If non-ancient blocks are partially migrated, the script will attempt to resume the migration.",
+	}
+	onlyAncientsFlag = &cli.BoolFlag{
+		Name:  "only-ancients",
+		Usage: "Use to only migrate ancient blocks. Ignored when running full migration",
```

Comment:
Do we need this flag? I can't think of a situation where it would be useful

Reply:
I found it useful for debugging / testing. Do you see any harm in keeping it?

Reply:
I thought that the plan was to run it with this flag a couple days prior the actual migration to migrate the majority of blocks in advance. How do you suggest we do that without the flag @piersy?

Reply:
Ok as discussed,  it's likely that people have snapshots that they can run this on prior to the actual migration and so won't need to stop their nodes, so given that scenario this flag does seem useful.

Reply:
Discussed in call, going to have validators pre-migrate from a snapshot

Reply:
And in answer to your question I'm pretty sure running this while the node is running has the potential to cause corruption.